smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaBatchRecords.java

```diff
@@ -0,0 +1,28 @@
+package io.smallrye.reactive.messaging.kafka;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.TopicPartition;
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+/**
+ * Represents a batch of Kafka records received by polling the {@link org.apache.kafka.clients.consumer.KafkaConsumer}
+ *
+ * This type extends the {@code Message<List<T>>} where {@code T} is the type of records' payloads.
+ * The complete list of Kafka record payloads are accessible via the {@link Message#getPayload()} method.
+ *
+ * @param <K> The record key type
+ * @param <T> The record payload type
+ */
+public interface KafkaBatchRecords<K, T> extends Message<List<T>>, Iterable<KafkaRecord<K, T>> {
```

Comment:
For symmetry with Kafka's `ConsumerRecord`/`ConsumerRecords`, should we perhaps call this `KafkaRecords`?

Reply:
Alternatively, `KafkaRecordBatch`.

`KafkaBatchRecords` seems disconnected from `KafkaRecord`.

Yes, I enjoy naming discussions :-)

Reply:
I was actually going to rename this to `KafkaRecordBatch` :) But I knew this discussion was coming. 
Kafka client internals (pre-deserialization) use `Record` and `RecordBatch`.
Lets go for `KafkaRecordBatch` ?

Reply:
+1